doc/format_specifications.rdoc

```diff
@@ -163,34 +176,123 @@ Format +argument+ as a binary integer:
 
   sprintf('%b', 1)  # => "1"
   sprintf('%b', 4)  # => "100"
+
   # Prefix '..' for negative value.
   sprintf('%b', -4) # => "..100"
 
-Type-specific modifier:
+  # Alternate format.
+  sprintf('%#b', 4)  # => "0b100"
+  sprintf('%#B', 4)  # => "0B100"
 
-- '#' flag' (use alternate format):
+==== Specifier +c+
 
-    sprintf('%#b', 1)  # => "0b1"
-    sprintf('%#B', 1)  # => "0B1"
-    sprintf('%#b', 4)  # => "0b100"
+Format +argument+ as a single character:
 
-==== Specifier +c+
+  sprintf('%c', 'A') # => "A"
+  sprintf('%c', 65)  # => "A"
 
 ==== Specifier +d+
 
+Format +argument+ as a decimal integer:
+
+  sprintf('%d', 100)  # => "100"
+  sprintf('%d', -100) # => "-100"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +e+ and +E+
 
+Format +argument+ in
+{scientific notation}[https://en.wikipedia.org/wiki/Scientific_notation]:
+
+  sprintf('%e', 3.14159)  # => "3.141590e+00"
+  sprintf('%E', -3.14159) # => "-3.141590E+00"
+
 ==== Specifier +f+
 
+Format +argument+ as a floating-point number:
+
+  sprintf('%f', 3.14159)  # => "3.141590"
+  sprintf('%f', -3.14159) # => "-3.141590"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +g+ and +G+
 
+Format +argument+ using exponential form (+e+/+E+ specifier)
+if the exponent is less than -4 or greater than or equal to the precision.
+Otherwise format +argument+ using floating-point form +f+ specifier).
+
+The precision defaults to 6, and can be changed to +n+ using the +.n+ flag.
```

Comment:
I think we should have examples involving changing the precision. For example:

```
sprintf('%.2g', 100) # => "1e+02"
```

Reply:
IIRC, @jeremyevans was adamantine about not mentioning unexceptional behavior of flags in the type-spec sections.  That principle would suggest not adding examples, but instead removing the last two lines cited above (I think).

Reply:
Now that we have a "Precision Specifier" section, I think this line can be removed.